qiskit/ignis/experiments/quantum_volume/qv_experiment.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Quantum volume Experiment.
+"""
+
+import copy
+import itertools
+import warnings
+
+import numpy as np
+import uuid
+import qiskit
+
+from qiskit.circuit.library import QuantumVolume
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+
+from typing import Optional, Dict, Union, List
+from qiskit.ignis.experiments.base import Experiment, Generator, Analysis
+
+from qiskit import transpile, assemble
+from qiskit.providers import BaseJob
+from qiskit.exceptions import QiskitError
+from qiskit.providers import BaseBackend
+
+
+class QuantumVolumeExperiment(Experiment):
+    """Quantum volume experiment."""
+
+    # pylint: disable=arguments-differ
+    def __init__(self,
+                 qubits: Union[int, List[int]],
+                 job: Optional = None,
+                 seed: Optional(int)=None):
+
+        if isinstance(qubits, int):
+            qubits = [range(qubits)]
+        self._qubits = qubits
+        self._trials = 0
+        self._seed = seed
+        self._circuits = [[] for _ in range(len(self._qubits))]
+        self._circuit_no_meas = [[] for _ in range(len(self._qubits))]
+        self._ideal_job = None
+        generator = QuantumVolumeGenerator(qubits)
+        analysis = QuantumVolumeAnalysis()
+        super().__init__(generator=generator, analysis=analysis, job=job)
+
+    @property
+    def circuits(self):
+        """Return experiment circuits"""
+        return self._circuits
+
+    @property
+    def circuit_no_meas(self):
+        """Return experiment circuits without measurement"""
+        return self._circuit_no_meas
+
+    @property
+    def qubits(self):
+        """Return experiment qubits"""
+        return self._qubits
+
+    @qubits.setter
+    def qubits(self, value):
+        """Set experiment qubits"""
+        if not isinstance(value, (List[int], int)):
+            raise QiskitError("Invalid qubits object. must be int or List[int]")
+        self._qubits = value
+
+    def add_data(self, trials: Optional(int)=1):
```

Comment:
in the base class `add_data` is in the `Analysis` class and not in the `Experiment` class.
Perhaps this function should be added to the base `Experiment` class? (since it's also useful for RB).